src/libraries/System.Private.CoreLib/src/System/Runtime/JitInfo.cs

```diff
@@ -0,0 +1,32 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.Runtime
+{
+    public static partial class JitInfo
+    {
+        private const long TicksPerMillisecond = 10000;
+        private const long TicksPerSecond = TicksPerMillisecond * 1000;
+
+        // "Frequency" stores the frequency of the high-resolution performance counter,
+        // if one exists. Otherwise it will store TicksPerSecond.
+        // The frequency cannot change while the system is running,
+        // so we only need to initialize it once.
+        public static readonly long Frequency = System.Diagnostics.Stopwatch.QueryPerformanceFrequency();
```

Comment:
You are making assumption that `System.Diagnostics.Stopwatch` is going to use same tick units as CoreCLR Win32 PAL. It is a fragile assumption to make.

It would be better if you have transformed the ticks to some fixed size units in coreclr/vm to avoid potential mismatch.

Reply:
I was hoping to avoid a call to `QueryPerformanceFrequency` on the vm side, but I can normalize it to 100ns ticks like mono does which should simplify the conversion on the managed end and remove ambiguity.

Reply:
> I was hoping to avoid a call to QueryPerformanceFrequency on the vm side,

Why? It is very cheap call.

Reply:
That's fair. I'm writing up a patch that does just that. Should be reusable in other places as well.